prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -387,7 +403,73 @@ def retrieve_top_datasets(
         dataset_names = [x.name for x in sorted_list]
         return dataset_names
 
-    def rerank_datasets(self, dataset_list: list[str], prompt_spec: PromptSpec):
+    def make_dataset_from_samples(
+        self,
+        inputs: list[str],
+        outputs: list[str],
+    ) -> datasets.DatasetDict:
+        """Given a list of inputs and outputs, make a dataset.
+
+        This function takes in inputs and outputs, both as list of strings,
+        and returns a DatasetDict object with a single split, "train". It has
+        two columns, "input_col" and "output_col".
+
+
+        Args:
+            inputs: A list of inputs, each input is a string.
+            outputs: A list of outputs, each output is a string.
+
+        Returns:
+            A DatasetDict object with a single split, "train". It has two
+            columns, "input_col" and "output_col".
+        """
+        updated_inputs, updated_outputs = [], []
+        if len(inputs) < 0 or len(inputs) != len(outputs):
+            logger.error("Length of inputs and outputs must be >0 and equal.")
+        else:
+            for i, o in zip(inputs, outputs):
+                if i is not None and o is not None:
+                    updated_inputs.append(i)
+                    updated_outputs.append(o)
+                else:
+                    logger.warning(f"Input or output is None: {i} {o}")
+
+        dataset_dict = {}
+        dataset_dict["train"] = datasets.Dataset.from_dict(
```

Comment:
I understand this function was earlier written in a way that it returned a dict where the train key had the dataset. If you think it does not change too much code otherwise, we should return the dataset directly instead of at `dataset_dict['train']`. But can do in another PR if you think this would require too many other downstream changes.

Reply:
Yeah, you make a fair point - I think the point of having the train split was to have consistency of datasets regardless of whether they were transformed or just retrieved (We only retrieve datasets which have a train split). Can add to a sep PR if reqd

Reply:
Let's do in a separate PR